### ISSUE TITLE
Cart Rule Maximum Amount

### ISSUE THREAD

**Issue by SharmPRO** [PrestaShop/PrestaShop#23012]
I need to apply a discount for an order of maximum 77€. Now the Cart Rule ask the minimum amount only.

The rule logic should consider  total >= minimum_amount && total <= maximum_amount

I'd considered to reverse the logic and to apply a negative fix discount but again fix discount value accept only positive value.

Thanks

---
#### Is your feature request related to a problem?
I need to apply a discount for an order of maximum 77€. Now the Cart Rule ask the minimum amount only.

#### Describe the solution you'd like
The rule logic should consider total >= minimum_amount && total <= maximum_amount

#### Alternatives you've considered
I'd considered to reverse the logic and to apply a negative fix discount but again fix discount value accept only positive value.

#### Additional context
N/A

**Comment by hibatallahAouadni** [comment]
Hello @SharmPRO 

Please **respect the provided template to submit a complete bug report**, we need these blocks (steps to reproduce, PrestaShop version, PHP version, ...) to reproduce your issue and to help you resolve your problem ASAP.

Thanks!

**Comment by SharmPRO** [comment]
Sorry but i wanted to fill a 'Feature Request'

**Comment by hibatallahAouadni** [comment]
Hello @SharmPRO 

So you should **respect the provided template to submit a feature** :wink: see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/106024813-70600a00-60c8-11eb-97a6-8efa0325f3fc.png)

Please edit your issue and add these blocks.

Thanks!

**Comment by SharmPRO** [comment]
#### Is your feature request related to a problem?

I need to apply a discount for an order of maximum 77€. Now the Cart Rule ask the minimum amount only.

#### Describe the solution you'd like

The rule logic should consider total >= minimum_amount && total <= maximum_amount

#### Alternatives you've considered

I'd considered to reverse the logic and to apply a negative fix discount but again fix discount value accept only positive value.

#### Additional context

N/A

Thanks

**Comment by hibatallahAouadni** [comment]
Hello @SharmPRO 

Thanks for your feedback and for the suggestion. 
The Product Team will take a look. Please stay tuned !

Ping @PrestaShop/product-team , is it possible to add maximum_amount as a condition for the cart rule?

**Comment by mrabsinthe** [comment]
Having possibility to set maximum amount would be great as it is currently not possible to set different gifts / discounts based on amount levels.

**Comment by partiid** [comment]
Yes, that would be great to implement. You could have a situtation where you have % discount and someone makes huuuge order. Then you have to give him the % but you would want voucher to be only for small orders (e.g new customers) 
Will it ever be considered to add  ?

**Comment by nikonhub** [comment]
Be able to apply a % discount only on the first product in a stack would be great. If 10 same products ordered the discount would be applied to the first product only.

**Comment by boo-code** [comment]
Still absent on 9.2: `ps_cart_rule` has `minimum_amount`, `minimum_amount_tax`, `minimum_amount_currency` and `minimum_amount_shipping`, and no maximum of any kind.

The reason this is worth pausing on rather than adding: the surface is mid-rewrite. 9.2 introduces `src/Core/Domain/Discount`, a CQRS layer whose handlers build a `CartRule` underneath (`AddDiscountHandler` calls `$this->discountBuilder->build($command)` and `$this->discountRepository->add($builtCartRule)`).

`AddDiscountCommand` currently models `minimumProductQuantity` and does **not** model `minimum_amount` at all. So the condition this issue wants a counterpart to has not been carried into the new command yet. Adding a maximum on the legacy side now would put it in `CartRule` and the form, while the replacement command still carries neither bound, and the two would have to be reconciled during the rework anyway.

The useful sequencing is the other way round: the amount conditions get modelled in `AddDiscountCommand`/`UpdateDiscountCommand`, and a maximum goes in at that point as a pair with the minimum rather than as a second legacy column.
